src/libs/SidebarUtils.js

```diff
@@ -93,86 +91,44 @@ Onyx.connect({
-        const hasAddWorkspaceRoomError = report.errorFields && !_.isEmpty(report.errorFields.addWorkspaceRoom);
-        const shouldFilterReportIfEmpty = report.lastMessageTimestamp === 0
-
-            // We make exceptions for defaultRooms and policyExpenseChats so we can immediately
-            // highlight them in the LHN when they are created and have no messsages yet. We do
-            // not give archived rooms this exception since they do not need to be higlihted.
-            && !(!ReportUtils.isArchivedRoom(report) && (isDefaultRoom || isPolicyExpenseChat))
-
-            // Also make an exception for workspace rooms that failed to be added
-            && !hasAddWorkspaceRoomError;
-
-        const shouldFilterReportIfRead = hideReadReports && !ReportUtils.isUnread(report);
-        const shouldFilterReport = shouldFilterReportIfEmpty || shouldFilterReportIfRead;
-        if (report.reportID !== currentlyViewedReportID
-            && !report.isPinned
-            && !hasDraftComment
-            && shouldFilterReport
-            && !reportContainsIOUDebt) {
-            return false;
-        }
-
-        // We let Free Plan default rooms to be shown in the App, or rooms that also have a Guide in them.
-        // It's the two exceptions to the beta, otherwise do not show policy rooms in product
-        if (ReportUtils.isDefaultRoom(report)
-            && !Permissions.canUseDefaultRooms(betas)
-            && ReportUtils.getPolicyType(report, policies) !== CONST.POLICY.TYPE.FREE
-            && !ReportUtils.hasExpensifyGuidesEmails(lodashGet(report, ['participants'], []))) {
-            return false;
-        }
-
-        if (ReportUtils.isUserCreatedPolicyRoom(report) && !Permissions.canUsePolicyRooms(betas)) {
-            return false;
-        }
+    // Get all the display names for our reports in an easy to access property so we don't have to keep
+    // re-running the logic
+    const filteredReportsWithReportName = _.map(filteredReports, (report) => {
+        const personalDetailMap = OptionsListUtils.getPersonalDetailsForLogins(report.participants, personalDetails);
+        return {
+            ...report,
+            reportDisplayName: ReportUtils.getReportName(report, personalDetailMap, policies),
+        };
+    });
 
-        if (isPolicyExpenseChat && !Permissions.canUsePolicyExpenseChat(betas)) {
-            return false;
+    // Sorting the reports works like this:
+    // - When in default mode, reports will be ordered by most recently updated (in descending order) so that the most recently updated are at the top
+    // - When in GSD mode, reports are ordered by their display name so they are alphabetical (in ascending order)
+    // - Regardless of mode, all archived reports should remain at the bottom
+    const orderedReports = _.sortBy(filteredReportsWithReportName, (report) => {
+        if (ReportUtils.isArchivedRoom(report)) {
+            return isInDefaultMode ? -Infinity : 'ZZZZZZZZZZZZZ';
```

Comment:
NABs, same question about `-Infinity` and why we would use it over `-1`

The `ZZZZZZZZZZZZZ` here also feels like a magic... word (since can't be a magic number 😂)  how did we come up with this and should we add a code comment so that the reasoning is a bit more clear?